eclass/kernel-build.eclass

```diff
@@ -427,12 +434,12 @@ kernel-build_merge_configs() {
 				CONFIG_MODULE_SIG_FORCE=y
 				CONFIG_MODULE_SIG_${MODULES_SIGN_HASH^^}=y
 			EOF
-			if [[ -e ${MODULES_SIGN_KEY} && -e ${MODULES_SIGN_CERT} &&
-				${MODULES_SIGN_KEY} != ${MODULES_SIGN_CERT} &&
-				${MODULES_SIGN_KEY} != pkcs11:* ]]
-			then
-				cat "${MODULES_SIGN_CERT}" "${MODULES_SIGN_KEY}" > "${T}/kernel_key.pem" || die
-				MODULES_SIGN_KEY="${T}/kernel_key.pem"
+			if [[ -n "${MODULES_SIGN_KEY_CONTENTS}" ]]; then
+				touch "${T}/kernel_key.pem" || die
+				chmod 0600 "${T}/kernel_key.pem" || die
```

Comment:
@vimproved did you see @mgorny 's comments about this on the mailing list? Could you address those? Then we can go for the second try at merging this fix.

Reply:
I went ahead and applied some small fixups to the patch and merged it. Thanks for the PR :+1: